### DISCUSSION TITLE
What do you think of re-licensing from GPL to AGPL?

### DISCUSSION THREAD

**Discussion by dashezup** [monkeytypegame/monkeytype#1534] [Questions]
monkeytype is currently server side software and licensed under GPL. But if someone uses your GPL licensed codes to provide a SaaS without open source it, you can't force them to open source it under the same license. And that's why there is AGPL.

https://tldrlegal.com/license/gnu-affero-general-public-license-v3-(agpl-3.0)

> It provides the same restrictions and freedoms as the GPLv3 but with an additional clause which makes it so that source code must be distributed along with web publication. Since web sites and services are never distributed in the traditional sense, the AGPL is the GPL of the web.

It may be worth mentioning a case regarding commercial open source, MongoDB is used to license under AGPL, but they re-licensed their codes to their own GPL-based license called SSPL to add more restriction a few years ago on the approach of copyleft. But SSPL is not OSI-approved license while AGPL is.

There are also a reason not to relicense, I saw that there are already a lot of contibutors of this project (195 shown). Relicensing may require every author of the project to agree to switch to the new license. IANAL. it could be time-consuming and may not worth the effort.

But if you are rewriting/replacing most parts or the whole of the project, you may consider this. Because it may be harder and harder in the further to change license.

**Comment by Miodec** [comment]
So basically with GPL people can host the same code with modifications without violating the license because they are not selling it correct?

Moving to AGPL sound like a good idea but as you said - do we need go get all contributors on board to do the switch "legally"? Are there any other requirements when changing licenses?

**Reply by dashezup** [reply]
> So basically with GPL people can host the same code with modifications without violating the license because they are not selling it correct?

No. it's not related to if it's commercial or not. it should be

> with GPL people can host the same code with modifications without violating the license because **they are running it on their server and provide it as a service to others (or just SaaS, software as a service) instead of traditional applications which distribute binaries build against the source code**

> do we need go get all contributors on board to do the switch "legally"

probably (such like open an issue and mention all of them, make a list to check if everyone agreed to switch to the new license). I don't have experience on re-licening either.

> Are there any other requirements when changing licenses?

not of any I can think of.

btw I just read this

https://opensource.org/faq#contributor-agreements

> What are "contributor agreements"? Are they like open source licenses?

> Many open source projects will only accept patches (code contributions or documentation contributions) from people who have submitted a legal document known as a contributor agreement. Contributor agreements are not open source licenses — rather, they are a way for the contributor to tell the project that it has the right to distribute the new contributions under the project's existing open source license. (**Some contributor agreements also allow for the project to distribute the contributions under other open source licenses too, which enables projects to change their license in the future**, and some agreements even allow the project to distribute the contributions under any license the project wants.)

(emphasis added)

https://en.wikipedia.org/wiki/Contributor_License_Agreement

> CLAs are important especially for corporate open source projects under a copyleft license, since without a CLA the contribution would restrict the guardian as well.

This project seems to be the case that it does not have contributor agreements and you need to ask every contributor to agree when you relicense it. If you could relicense it, you should add contributor agreements as well imo.

as always, IANAL, TINLA.
